### PR TITLE
DLS-8805: Make trading name compulsary

### PR DESCRIPTION
**Pull request by RaminEsfandiari**

### CODE REVIEW

File: app/controllers/actions/ControllerActionHelper.scala
Comment (GinnyThomas) [inline app/controllers/actions/ControllerActionHelper.scala:1]: Can you separate these methods out further?  Possibly by subtask/purpose - for example the trading name - can you break down the packaging site case match into a separate method and then call it here, same with warehouses

**Review by GinnyThomas — DISMISSED** [review]

File: app/controllers/addressLookupFrontend/RampOffController.scala [hunk 1]
```diff
@@ -52,23 +55,43 @@ class RampOffController @Inject()(identify: IdentifierAction,
...
     implicit request =>
+      val userAnswers = request.userAnswers
       for {
         alfResponse         <- addressLookupService.getAddress(alfId)
-        updatedUserAnswers = addressLookupService.addAddressUserAnswers(WarehouseDetails, alfResponse.address, request.userAnswers, sdilId, alfId)
+        ukAddress           = addressLookupService.addressChecker(alfResponse.address, alfId)
+        optTradingName      = alfResponse.address.organisation
+        updatedUserAnswers = optTradingName match {
+          case Some(tradingName) => userAnswers.addWarehouse(ukAddress, tradingName, sdilId)
+          case None => userAnswers.setAlfResponse(ukAddress, WarehouseDetails, sdilId)
```
[the remark this answers is not in the document]
Comment (Chrissystk) [reply app/controllers/addressLookupFrontend/RampOffController.scala:65]: The SdilId and AlfId are different. The alfId is part of the actual address where the sdilId is used for the entry into the Map. It is the name of the id for the url.

File: app/controllers/actions/ControllerActionHelper.scala [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            genericLogger.logger.info(s"User has no user answers ${hc.requestId}")
+            Future.successful(Left(Redirect(routes.RegistrationController.start)))
+        }
+      }
+
+      override protected def executionContext: ExecutionContext = ec
+    }
+  }
+
+  def dataRequiredForUserWhoCannotRegisterAction(registerState: RegisterState)(implicit ec: ExecutionContext): ActionRefiner[OptionalDataRequest, DataRequest] = {
```
[the remark this answers is not in the document]
Comment (Chrissystk) [reply app/controllers/actions/ControllerActionHelper.scala:64]: This isn't new. Just moved to a trait as the ControllerAction was getting big. This is an action specific to users who need to signin again, enter an utr or are already registered (i.e users who cannot register (or users who cannot register until they provide a valid utr)).

File: app/controllers/actions/ControllerActionHelper.scala
Comment (drewmiley) [reply app/controllers/actions/ControllerActionHelper.scala:1]: In this instance, I agree with Christine that it is perhaps slightly outside the scope of this PR - considering the size of it and the business impact. I do agree that if interacting with this code in future, we should improve it as you suggested.

File: app/controllers/actions/ControllerActionHelper.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (drewmiley) [reply app/controllers/actions/ControllerActionHelper.scala:64]: In this instance, I agree with Christine that it is perhaps slightly outside the scope of this PR - considering the size of it and the business impact. I do agree that if interacting with this code in future, we should improve it as you suggested.

File: app/controllers/addressLookupFrontend/RampOffController.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (drewmiley) [reply app/controllers/addressLookupFrontend/RampOffController.scala:65]: There is possibly a case for renaming the`sdilId`, but as a pre-existing name - and considering the size and the scope of the PR - I would leave that potential refactor for a smaller ticket where the impact would not be as uncertain

**Review by drewmiley — APPROVED** [review]

**Review by drewmiley — APPROVED** [review]

**Review by GinnyThomas — APPROVED** [review]

**Pull request merged**